drivers/include/lis2dh12.h

```diff
@@ -114,13 +114,100 @@ enum {
 };
 
+/**
+ * @brief   Interrupt config register values
+ */
+enum {
+    LIS2DH12_INT_CFG_XLIE = 0x01,   /**< enable X low event or direction recognition */
+    LIS2DH12_INT_CFG_XHIE = 0x02,   /**< enable X high event or direction recognition */
+    LIS2DH12_INT_CFG_YLIE = 0x04,   /**< enable Y low event or direction recognition */
+    LIS2DH12_INT_CFG_YHIE = 0x08,   /**< enable Y high event or direction recognition */
+    LIS2DH12_INT_CFG_ZLIE = 0x10,   /**< enable Z low event or direction recognition */
+    LIS2DH12_INT_CFG_ZHIE = 0x20,   /**< enable Z high event or direction recognition*/
+    LIS2DH12_INT_CFG_6D   = 0x40,   /**< enable 6-direction detection */
+    LIS2DH12_INT_CFG_AOI  = 0x80,   /**< and/or combination of interrupt events */
+};
+
+/**
+ * @brief   Interrupt type values
+ */
+enum {
+    /* for interrupt 1 (CTRL_REG3) */
+    LIS2DH12_INT_1_TYPE_OVERRUN = 0x02, /**< FIFO overrun interrupt on INT_1 */
+    LIS2DH12_INT_1_TYPE_WTM     = 0x04, /**< FIFO watermark interrupt on INT_1 */
+    LIS2DH12_INT_1_TYPE_ZYXDA   = 0x10, /**< ZYXDA interrupt on INT_1 */
+    LIS2DH12_INT_1_TYPE_IA2     = 0x20, /**< IA2 interrupt on INT_1 */
+    LIS2DH12_INT_1_TYPE_IA1     = 0x40, /**< IA1 interrupt on INT_1 */
+    LIS2DH12_INT_1_TYPE_CLICK   = 0x80, /**< click interrupt on INT_1 */
+
+    /* for interrupt 2 (CTRL_REG6) */
+    LIS2DH12_INT_2_TYPE_POLARITY = 0x02, /**< INT_1 and INT_2 pin polarity (0: high-active; 1: low-active) */
+    LIS2DH12_INT_2_TYPE_ACT      = 0x08, /**< enable activity interrupt on INT_2 */
+    LIS2DH12_INT_2_TYPE_BOOT     = 0x10, /**< enable boot on INT_2 */
+    LIS2DH12_INT_2_TYPE_IA2      = 0x20, /**< IA2 on INT_2 */
+    LIS2DH12_INT_2_TYPE_IA1      = 0x40, /**< IA1 on INT_2 */
+    LIS2DH12_INT_2_TYPE_CLICK    = 0x80, /**< click interrupt on INT_2 */
+};
+
+/**
+ * @brief    Parameter for interrupt configuration
+ */
+typedef struct {
+    uint8_t int_config;             /**< values for configuration */
+    uint8_t int_threshold:7;        /**< the threshold for triggering interrupt, threshold in range 0-127 */
+    uint8_t int_duration:7;         /**< time between two interrupts, calc from <duration>/ODR, ODR section in CTRL_REG1, duration in range 0-127 */
+    uint8_t int_type;               /**< values for type of interrupts */
+} lis2dh12_int_params_t;
+
+/**
+ * @brief   Status of INT_SRC register
+ */
+typedef union {
+    struct {
+        uint8_t LIS2DH12_INT_SRC_XL:1;  /**< X low event has occured */
+        uint8_t LIS2DH12_INT_SRC_XH:1;  /**< X high event has occured */
+        uint8_t LIS2DH12_INT_SRC_YL:1;  /**< Y low event has occured */
+        uint8_t LIS2DH12_INT_SRC_YH:1;  /**< Y high event has occured */
+        uint8_t LIS2DH12_INT_SRC_ZL:1;  /**< Z low event has occured */
+        uint8_t LIS2DH12_INT_SRC_ZH:1;  /**< Z high event has occured */
+        uint8_t LIS2DH12_INT_SRC_IA:1;  /**< 1 if interrupt occured */
+    };
+    uint8_t LIS2DH12_INT_SRC;
```

Comment:
please use lowercase for variables.
Also no need to prefix struct members with `LIS2DH12`.